sw/common/gpio.c

```diff
@@ -4,28 +4,29 @@
 
 #include "gpio.h"
 #include "dev_access.h"
+#include "demo_system.h"
 
 void set_outputs(gpio_t gpio, uint32_t outputs) {
   DEV_WRITE(gpio, outputs);
 }
 
-uint32_t get_outputs(gpio_t gpio) {
+uint32_t read_gpio(gpio_t gpio) {
```

Comment:
The intent of the `gpio` is to provide a pointer to the gpio base, rather than a specific GPIO register. So we should add the relevant register offset here for the read. Then you could add a second parameter to choose debounced or non-debounced or just have two separate read functions.

Reply:
Sorry @GregAC, I couldn't understand what is your suggestion here.